File: app/src/main/java/com/jetpack/xhb/okio/Source.java
package com.jetpack.xhb.okio;

import java.io.Closeable;
import java.io.IOException;

/**
 * 读 输入流
 */
public interface Source extends Closeable {

    long read(Buffer sink, long byteCount) throws IOException;

    Timeout tiemout();

    @Override void close() throws IOException;
}

File: app/src/main/java/com/jetpack/xhb/okio/kotlin/Source.kt
package com.jetpack.xhb.okio.kotlin

import java.io.Closeable
import java.io.IOException

interface Source : Closeable {

    @Throws(IOException::class)
    fun read(sink: Buffer, byteCount: Long): Long

    fun timeout(): Timeout

    @Throws(IOException::class)
    override fun close()

}